jcl/src/java.base/share/classes/java/security/AccessController.java

```diff
@@ -500,7 +504,9 @@ private static AccessControlContext getContextHelper(boolean forDoPrivilegedWith
 		}
 		accLower = accTmp;
 	}
-
+	if (accContext != null) {
+		accContext.domainCombiner = dc;
```

Comment:
Is accContext.domainCombiner expected to be null here?

Reply:
`dc` could be `null`.
Did you mean `accContext` instead? It is not expected to be `null` but the code structure could let it stay `null` if `frameNbr` is zero. (Eclipse has a warning about this).

Reply:
I'm wondering if dc can overwrite the domainCombiner already contained in the accContext.domainCombiner

Reply:
`dc` is the first `non-null domainCombiner` discovered so it shouldn't overwrite a different `non-null domainCombiner`. To be safe, adding a `null` check for this assignment.